docs/guides/custom_auth_action.md

```diff
@@ -0,0 +1,100 @@
+
+Because of its distinct differences from regular Data Actions, a special resource has been defined to manage the Custom Auth Data Action and its configuration: `genesyscloud_integration_custom_auth_action`.
+
+In this guide we will see examples on how to manage the resource and its behavior on different cases.
+
+## Requirements
+
+The existence of the Custom Auth Data Action is completely managed by Genesys Cloud. Before configuring it with CX as Code, you would need:
+
+1. A Genesys Cloud Web Services Data Actions integration.
+2. The integration should be configured with `User Defined (OAuth)` credentials type.
+
+It's best practice to manage the integration, credentials, and custom auth action all on CX as Code but it's also possible to manage only the `genesyscloud_integration_custom_auth_action` on an existing integration. The requirements still have to be met for it to work though.
+
+## Managing the integration and its custom auth action in CX as Code
+
+The following snippet is an example of how you can manage your integration infrastructure along with the credentials type and custom auth data action:
+
+```hcl
+resource "genesyscloud_integration_credential" "credential" {
+  name                 = "example-credential"
+  credential_type_name = "userDefinedOAuth"
+  fields = {
+    clientId = "groot"
+    clientSecret = "iamgroot"
+  }
+}
+
+resource "genesyscloud_integration" "example_integration" {
+  intended_state   = "ENABLED"
+  integration_type = "custom-rest-actions"
+  config {
+    name = "Example Integration Name"
+    credentials = {
+      basicAuth = genesyscloud_integration_credential.credential.id
+    }
+  }
+}
+
+resource "genesyscloud_integration_custom_auth_action" "auth_action" {
+  integration_id = genesyscloud_integration.example_integration.id
+  config_request {
+  request_url_template = "https://example-domain.com/loginurl"
+    request_type         = "POST"
+    request_template     = "grant_type=client_credentials"
+    headers = {
+      Authorization = "Basic $encoding.base64(\"$${credentials.clientId}:$${credentials.clientSecret}\")"
+    }
+  }
+  config_response {
+    success_template = "$${rawResult}"
+  }
+}
+```
+
+Take note of the configurations of both `genesyscloud_integration_credential` and the `genesyscloud_integration`: the credential should be of type `userDefinedOAuth`, and the integration should be of type `custom-rest-actions`. Having values other than these, on any of the two resources, would result in failure as Genesys Cloud will not create the Custom Auth Action and CX as Code wouldn't have any `genesyscloud_integration_custom_auth_action` resource to manage.
+
+## 'Deleting' the resource from CX as Code
+
+As mentioned, it is not possible to delete the Custom Auth Data Action from CX as Code as its existence is managed by Genesys Cloud. If you want to delete the Custom Auth Action, then you must also delete the integration it is associated with.
```

Comment:
If the credential type configured on the integration changes to a type other than "User Defined (Oauth)", then the custom auth action is deleted

Reply:
Added